server/server_test.go

```diff
@@ -37,35 +37,35 @@ func init() {
 	}
 }
 
-func TestOpenFGAServer(t *testing.T) {
+func TestServerWithPostgresDatastore(t *testing.T) {
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "postgres")
 
-	t.Run("TestPostgresDatastore", func(t *testing.T) {
-		testDatastore := storagefixtures.RunDatastoreTestContainer(t, "postgres")
+	uri := testDatastore.GetConnectionURI()
+	ds, err := postgres.NewPostgresDatastore(uri)
+	require.NoError(t, err)
 
-		uri := testDatastore.GetConnectionURI()
-		ds, err := postgres.NewPostgresDatastore(uri)
-		require.NoError(t, err)
-		defer ds.Close()
+	test.RunAllTests(t, ds)
+}
 
-		test.RunAllTests(t, ds)
-	})
+func TestServerWithMemoryDatastore(t *testing.T) {
+	ds := memory.New(telemetry.NewNoopTracer(), 10, 24)
+	test.RunAllTests(t, ds)
+}
 
-	t.Run("TestMemoryDatastore", func(t *testing.T) {
-		ds := memory.New(telemetry.NewNoopTracer(), 10, 24)
-		defer ds.Close()
-		test.RunAllTests(t, ds)
-	})
+func TestServerWithMySQLDatastore(t *testing.T) {
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
 
-	t.Run("TestMySQLDatastore", func(t *testing.T) {
-		testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
+	uri := testDatastore.GetConnectionURI()
+	ds, err := mysql.NewMySQLDatastore(uri)
+	require.NoError(t, err)
 
-		uri := testDatastore.GetConnectionURI()
-		ds, err := mysql.NewMySQLDatastore(uri)
-		require.NoError(t, err)
-		defer ds.Close()
+	test.RunAllTests(t, ds)
+}
 
-		test.RunAllTests(t, ds)
-	})
+func TestOpenFGAServer(t *testing.T) {
```

Comment:
Don't think this is needed anymore. Let me double check that we're not referencing it externally somehow..

Reply:
Yep, not needed. We only use `RunAllTests` externally.